### PR TITLE
Additional keyboard navigation in formbuilder

### LINKED ISSUE #555: Default options can be replaced with single click
User should be able to replace default option text without having to highlight or delete it (like placeholders)

### PR DESCRIPTION
**Pull request by duvld**
## Description

Main thread: https://www.flowdock.com/app/kobotoolbox/kobo/threads/a571W3FWGd8-BDcwwsK9uPT9EO2

~~PR to https://github.com/kobotoolbox/kpi/tree/2152-drop-makeEditable because these changes depend on using `input` fields instead of `makeEditable` fields. Made seperate PR to differentiate the two branches.~~

⚠️ Note: May have conflicts with eventual PR to add ODKRank type (https://github.com/kobotoolbox/kpi/issues/2420) in this branch: https://github.com/kobotoolbox/kpi/tree/2420-koborank-odkrank-model due to changes in `view.choices.coffee`.

### Changes to be made
Suggested changes: https://github.com/kobotoolbox/kpi/issues/553#issuecomment-614078698

- [x] ENTER and Arrow Key navigation for selecting question
- [x] ENTER navigation for choice questions, pressing ENTER should add new option
- [x] Navigation between labels through tabbing
- [x] ~~Shortcut for adding a new question~~ Navigate to `+` option through tabbing
- [ ] SHIFT-TAB should also show `+` button above last question row
    - Currently not shown due to having `display: none` if not moused over

## One Tab to Rule Them All
Use TAB for navigation across formbuilder.

Current implementation of this user story below:

1. User clicks on [+] to create new question
2. Writes: What is your name? [enter]
3. Presses [right arrow] 3x, [enter] to select text question
4. [+] is highlighted, showing it can be activated with [enter]
5. Presses [enter] > new question box appears
6. Writes: Are you currently displaced? [enter]
7. Presses [enter] to select single select question
8. (first response option is highlighted, meaning user can type to replace placeholder)
9. Writes: Yes [enter]
10. Writes: No [enter]
11. (third response option is highlighted) User doesn't want to add more options, presses [Tab]
12. [+] is highlighted to add another question

553-navigation-story
Added boxes around each tabbable element:
553-tabbing

## Related issues
"ENTER navigation for choice questions, pressing ENTER should add new option" fixes #555 
May have conflicts with eventual PR to add ODKRank type #2420
Part of #553

### CODE REVIEW

File: jsapp/scss/components/_kobo.form-builder.scss [hunk 1]
```diff
@@ -343,6 +374,17 @@ $t-form-builder-aside-open: 200ms;
...
     }
+
+    &:focus {
+      outline-color: rgb(77, 144, 254);
+      outline-offset: -2px;
+      outline-style: auto;
+      outline-width: 5px;
+
+      i {
+      outline-color: transparent;
```
Comment (magicznyleszek) [inline jsapp/scss/components/_kobo.form-builder.scss:385]: indentation

**Comment by duvld** [comment]
> I've found few issues:
> 
> 1. I think that tabbing into `+` to add a question, but only if it is visible, makes it a poor feature. I would  either work more on this here, or create a separete PR to improve it. Maybe make the `+` appear both after and before a question/group if something is focused inside the question/group? Additional complexity happens with nested groups!
> 2. There is a difference between mouse clicking `+` and ENTER activating it:
>    
>    1. MOUSE Screenshot 2020-06-03 at 15 29 59
>    2. ENTER Screenshot 2020-06-03 at 15 29 47
> 3. When adding a question with keyboard, the `+` that appears focused, appears at the end of the form, not just after a question. This can be observed if I have 10 questions, but I add new one at the top, and then `+` appears at the end not just after the one I've added. Similar problem when I keyboard add a question inside a group, I end up with `+` focused outside the group.

For 1. I think I know about this issue and my plan is to approach it with this:
> SHIFT-TAB should also show + button above last question row
> Currently not shown due to having display: none if not moused over

Let me know if this is what you're talking about/what you think :)

For 2. and 3., sometimes old issues are still present for some reason. I think fixed both of these already and they work fine for me on the staging server I set up for this PR: https://kf.tri.kbtdev.org have you tried to recreate them on there as well?

**Comment by magicznyleszek** [comment]
I've found a great css selector that fixes the problems I had with `+` button: https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-within and generally I think this is ready to be merged :ship::it: @duvld do you have any objections to merging this?

**Comment by duvld** [comment]
@magicznyleszek great fix for the `+` button functionality :) functionally I think we're good to go, I did have an issue with the styling on the right side header buttons when they're tabbed:
image
I'm going to try to fix this quickly by removing outlines when focused

**Pull request merged**